docs/BUILD_AND_TEST.md

```diff
@@ -61,72 +62,47 @@ Alternatively, you can run `make` with any of the following targets:
+Our tests are written using [smartystreets' GoConvey package](https://github.com/smartystreets/goconvey).  See https://github.com/smartystreets/goconvey/wiki for an introduction to creating a test using this package.
 
 ### Running Tests
 #### In local machine
 First you need to run the following to get all test dependencies (this runs ./scripts/test.sh as well):
 ```
-make test
+$ cd $GOPATH/src/github.com/intelsdi-x/snap
+$ make test
 ```
+
 To run all tests in order (it will stop at any failure in a directory before continuing on):  
 ```
-./scripts/test.sh
+$ ./scripts/test.sh
 ```
+
 To run all tests *and to continue through all directories even with errors*:  
 ```
-go test ./...
-```  
-Tests can be pruned in go test with the `-run` option:
-```
-go test ./... -v -run <TestName>
-```
-
-e.g. `TestLoad` from `control_test.go`:
+$ go test ./...
+``` 
+ 
+Tests can be pruned in go test with the `-run` option followed by the test name (e.g. `TestLoad` from `control_test.go`):
 ```
-go test ./... -v -run TestLoad
+$ go test ./... -v -run TestLoad
 ```
 
-e.g. using GoConvey UX:
+Or using GoConvey ([in a browser](https://github.com/smartystreets/goconvey#in-the-browser)):
 ```
-go test -coverprofile=/tmp/coverage.out && go tool cover -html=/tmp/coverage.out
+$ go test -coverprofile=/tmp/coverage.out && go tool cover -html=/tmp/coverage.out
 ```
 
 #### In Docker
-You need to have Docker Engine installed. Visit [Install Docker Engine](https://docs.docker.com/engine/installation/) for detailed instructions on how to do it.
-If you're using Docker on OS X(Darwin) you need to have an active Docker host and the env var, `$DOCKER_HOST`, needs to be exported. 
-
-There's a `Dockerfile` located at `./scripts/Dockerfile`:
-```
-FROM golang:latest  
-ENV GOPATH=$GOPATH:/app
-ENV SNAP_PATH=/go/src/github.com/intelsdi-x/snap/build
-RUN apt-get update && \
-    apt-get -y install facter
-WORKDIR /go/src/github.com/intelsdi-x/
-RUN git clone https://github.com/intelsdi-x/gomit.git
-WORKDIR /go/src/github.com/intelsdi-x/snap
-ADD . /go/src/github.com/intelsdi-x/snap
-RUN go get github.com/tools/godep && \
-    go get golang.org/x/tools/cmd/goimports && \
-    go get golang.org/x/tools/cmd/cover && \
-    go get github.com/smartystreets/goconvey
-RUN scripts/deps.sh
-RUN make
-```
-This is run in the snap directory using `./scripts/run_tests_with_docker.sh`  
-```
-#!/bin/bash -e
-
-docker build -t intelsdi-x/snap-test -f scripts/Dockerfile .
-docker run -it intelsdi-x/snap-test scripts/test.sh
+The Snap Framework supports running tests in an isolated container as opposed to your local host. You need to have Docker Engine installed to do so. Visit [Install Docker Engine](https://docs.docker.com/engine/installation/) for detailed instructions on how to do it. If you're using Docker on OS X(Darwin) you need to have an active Docker host and the `env` var, `$DOCKER_HOST`, needs to be exported.
```

Comment:
You wrote it @tiffanyfj 😄. I'm tempted to delete this part all together - do you think developers will want to deploy a container to test?